microcosm_pubsub/tests/chain/test_statements.py

```diff
@@ -0,0 +1,160 @@
+from hamcrest import (
+    assert_that,
+    calling,
+    equal_to,
+    is_,
+    raises,
+)
+
+from microcosm_pubsub.chain import Chain
+from microcosm_pubsub.chain.statements import extract, if_, switch, try_chain
+
+
+class TestStatements:
+
+    def test_extract(self):
+        chain = Chain(
+            extract("param", "arg"),
+            lambda param: param,
+        )
+        assert_that(
+            chain.resolve(arg=200),
+            is_(equal_to(200)),
+        )
+
+    def test_extractreturn_value(self):
+        chain = Chain(
+            extract("param", "arg"),
+        )
+        assert_that(
+            chain.resolve(arg=200),
+            is_(equal_to(200)),
+        )
+
+    def test_extractwith_propery(self):
+        chain = Chain(
+            extract("param", "arg", "data"),
+        )
+        assert_that(
+            chain.resolve(arg=dict(data=200)),
+            is_(equal_to(200)),
+        )
+
+    def test_extractwith_attribute(self):
+        chain = Chain(
+            extract("param", "arg", "__class__"),
+        )
+        assert_that(
+            chain.resolve(arg=dict()),
+            is_(equal_to(dict)),
+        )
+
+    def test_if_(self):
+        chain = Chain(
+            if_("arg", chain=Chain(lambda: 200), other=Chain(lambda: 400)),
```

Comment:
A more fluent syntax for me would be:

```
when("arg").then(
   something,
).otherwise(
   something_else,
)
```

You just need `when` and `otherwise` to be builder functions that return the `When` object after updating internal state.